src/CSnakes.Runtime/ServiceCollectionExtensions.cs

```diff
@@ -179,6 +179,18 @@ public static IPythonEnvironmentBuilder FromConda(this IPythonEnvironmentBuilder
         return builder;
     }
 
+    public static IPythonEnvironmentBuilder FromManagedPython(this IPythonEnvironmentBuilder builder)
+    {
+        builder.Services.AddSingleton<PythonLocator>(
+            sp =>
+            {
+                var logger = sp.GetRequiredService<ILogger<ManagedPythonLocator>>();
+                return new ManagedPythonLocator(logger);
+            }
+        );
```

Comment:
```suggestion
        builder.Services.AddSingleton<PythonLocator, ManagedPythonLocator>();
```

Registers the type `ManagedPythonLocator` to be resolved as `PythonLocator`.